provider-blackduck/src/main/java/com/synopsys/integration/alert/provider/blackduck/processor/message/service/policy/BlackDuckComponentPolicyDetailsCreator.java

```diff
@@ -30,11 +32,12 @@ public BlackDuckComponentPolicyDetailsCreator(BlackDuckPolicySeverityConverter p
         this.policySeverityConverter = policySeverityConverter;
     }
 
-    public ComponentPolicy toComponentPolicy(ComponentPolicyRulesView componentPolicyRulesView) {
+    public ComponentPolicy toComponentPolicy(ComponentPolicyRulesView componentPolicyRulesView, Optional<PolicyRuleView> policyRuleView) {
```

Comment:
Taking something optional as a method parameter is ill-advised. I suspect if you moved the policy retrieval logic into this class this could be removed.